### PR TITLE
fix: dummy paymaster and data addresses updates

### PR DESCRIPTION
**Pull request by avarobinson**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
The focus of this PR is to update Alchemy paymaster addresses based on specific chains and IDs.

### Detailed summary
- Added `polygonMumbai` and `sepolia` to gas manager
- Updated Alchemy paymaster addresses for different chains and IDs

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/alchemy/src/gas-manager.ts [hunk 1]
```diff
@@ -6,40 +6,50 @@ import {
...
+      return AlcehmyPaymasterAddressV3;
     case arbitrumSepolia:
+      return ArbSepoliaPaymasterAddress;
+    case sepolia:
+    case goerli:
+    case polygonMumbai:
       return OldAlchemyPaymasterAddress;
     default:
-      return ArbSepoliaPaymasterAddress;
+      return OldAlchemyPaymasterAddress;
```
Comment (rthomare) [inline packages/alchemy/src/gas-manager.ts:53]: I think at this point we should throw... we shouldn't put a default value in the enum.

File: packages/alchemy/src/gas-manager.ts [hunk 2]
```diff
@@ -6,40 +6,50 @@ import {
...
     case zora:
     case zoraSepolia:
     case fraxtal:
     case fraxtalTestnet:
-      return NewAlchemyPaymasterAddress;
+      return AlcehmyPaymasterAddressV3;
     case arbitrumSepolia:
+      return ArbSepoliaPaymasterAddress;
+    case sepolia:
+    case goerli:
```
Comment (avasisht23) [inline packages/alchemy/src/gas-manager.ts:49]: we shouldn't support eth goerli anymore right? we deprecated support

File: packages/alchemy/src/gas-manager.ts [hunk 3]
```diff
@@ -6,40 +6,50 @@ import {
...
+  polygonMumbai,
+  sepolia,
   zora,
   zoraSepolia,
 } from "viem/chains";
 
-export const NewAlchemyPaymasterAddress =
+export const AlchemyPaymasterAddressV2 =
   "0x4Fd9098af9ddcB41DA48A1d78F91F1398965addc";
 export const OldAlchemyPaymasterAddress =
```
Comment (avasisht23) [inline packages/alchemy/src/gas-manager.ts:23]: call this v1?

File: packages/alchemy/src/gas-manager.ts (hunk 1 above)
Comment (rthomare) [reply packages/alchemy/src/gas-manager.ts:53]: If it's a chain we don't support no use in putting a random address in there.

File: packages/alchemy/src/gas-manager.ts [hunk 4]
```diff
@@ -6,40 +6,50 @@ import {
...
   "0xc03aac639bb21233e0139381970328db8bceeb67";
 export const ArbSepoliaPaymasterAddress =
   "0x0804Afe6EEFb73ce7F93CD0d5e7079a5a8068592";
+export const AlcehmyPaymasterAddressV3 =
```
Comment (avasisht23) [inline packages/alchemy/src/gas-manager.ts:27]:
```suggestion
export const AlchemyPaymasterAddressV3 =
```

File: packages/alchemy/src/gas-manager.ts (hunk 1 above)
Comment (rthomare) [reply packages/alchemy/src/gas-manager.ts:53]: Also it's probably better to match on chainId than just the whole chain object. That way we can support customized chains.

**Review by rthomare — COMMENTED** [review]
Not going to block on my comments but we should make some updates here.

File: packages/alchemy/src/gas-manager.ts (hunk 3 above)
Comment (avasisht23) [reply packages/alchemy/src/gas-manager.ts:23]: also nit: can we order this list in descending order: v3, v2, v1

File: packages/alchemy/src/gas-manager.ts [hunk 5]
```diff
@@ -6,40 +6,50 @@ import {
...
     case fraxtalTestnet:
-      return NewAlchemyPaymasterAddress;
+      return AlcehmyPaymasterAddressV3;
     case arbitrumSepolia:
+      return ArbSepoliaPaymasterAddress;
+    case sepolia:
+    case goerli:
+    case polygonMumbai:
       return OldAlchemyPaymasterAddress;
     default:
```
Comment (avasisht23) [inline packages/alchemy/src/gas-manager.ts:52]: don't need to have default different from the above cases

File: packages/alchemy/src/gas-manager.ts (hunk 1 above)
Comment (moldy530) [reply packages/alchemy/src/gas-manager.ts:53]: we need to update these middleware factories to also allow for paymaster address and dummy paymaster and data overrides. that way we're not blocking devs on new pushes

**Review by moldy530 — APPROVED** [review]

**Pull request merged**